hapi-fees/src/main/java/com/hedera/services/usage/crypto/ExtantCryptoContext.java

```diff
@@ -191,16 +194,24 @@ public ExtantCryptoContext build() {
 			return this;
 		}
 
-		public ExtantCryptoContext.Builder setCurrentNftAllowances(List<GrantedNftAllowance> currentNftAllowances) {
-			this.currentNftAllowances = convertToNftMapFromGranted(currentNftAllowances);
+		public ExtantCryptoContext.Builder setCurrentApproveForAllNftAllowances(List<GrantedNftAllowance> currentApproveForAllNftAllowances) {
+			this.currentApproveForAllNftAllowances = convertToNftMapFromGranted(currentApproveForAllNftAllowances);
 			mask |= NFT_ALLOWANCES_MASK;
 			return this;
 		}
 	}
+}
 
-	public record AllowanceMapKey(Long tokenNum, Long spenderNum) {
-	}
+record AllowanceId(Long tokenNum, Long spenderNum) implements Comparable<AllowanceId>{
 
-	public record AllowanceMapValue(Boolean approvedForAll, List<Long> serialNums) {
+	@Override
+	public int compareTo(@NotNull final AllowanceId that) {
+		return new CompareToBuilder()
+				.append(tokenNum, that.tokenNum)
+				.append(spenderNum, that.spenderNum)
+				.toComparison();
 	}
 }
+
+record AllowanceDetails(Boolean approvedForAll, List<Long> serialNums) {
+}
```

Comment:
Will be moved out and made public, since they are used in multiple classes in the delete allowance PR

Reply:
sounds good.